### ISSUE TITLE
[Performance]Re-Styling takes too long

### ISSUE THREAD

**Issue by PieroCastillo** [AvaloniaUI/Avalonia#5667]
Re-Style the controls takes too long.
I don't know exactly the causes, but I made a test and the method https://github.com/AvaloniaUI/Avalonia/blob/5e61abdb9e371890cf2fc473749a5bd3c3626185/src/Avalonia.Styling/StyledElement.cs#L333 was executed more than 2k times.(it was not a formal test but you can have an idea)

- video with [Aura.UI Sample](https://github.com/PieroCastillo/Aura.UI/tree/master/samples/Aura.UI.Sample) at run-time with RELEASE config in net 5 (avalonia 0.10 stable)

https://user-images.githubusercontent.com/39932324/111256727-c12fb180-85e7-11eb-9692-20b50ca5bf04.mp4

- video with  Windows UI Controls Gallery (downloaded from MS store)

https://user-images.githubusercontent.com/39932324/111256805-e45a6100-85e7-11eb-88ec-60ade2e0be55.mp4

Common settings:

- Windows 10 20H2
- Ryzen 5 5600X
- GTX 1660ti

I would know if there is some solution, that's all :)

**Comment by grokys** [comment]
Duplicate of https://github.com/AvaloniaUI/Avalonia/issues/4655.

tldr; Yes, we know that switching themes is currently slow. We've improved it a bit but there's more to do.

Going to close this as we'll track the work in #4655.
